recipe/conda_build_config.yaml

```diff
@@ -110,6 +119,9 @@ channel_targets:
 
 docker_image:                                   # [linux]
   - condaforge/linux-anvil-comp7                # [linux64]
+  - condaforge/linux-anvil-cuda:9.2             # [linux64]
+  - condaforge/linux-anvil-cuda:10.0            # [linux64]
+  - condaforge/linux-anvil-cuda:10.1            # [linux64]
```

Comment:
Which one of these is used when cuda is not used?

Reply:
Should be `condaforge/linux-anvil-comp7`. Though probably more tweaking is needed to get that to work correctly.

Reply:
Can you try rerendering an existing recipe without cuda to see what happens?

Reply:
I have previously and recall there were issues, but I don't recall what they were. Are you interested in knowing exactly how it fails? If so, I can put together an example.

Reply:
> Are you interested in knowing exactly how it fails? 

Yes

Reply:
Sure. Generated PR ( https://github.com/conda-forge/zlib-feedstock/pull/35 ), which shows what happens. It still works in the sense that the image selected should still perform the build. However it seems to still be using a CUDA image for a non-CUDA build, which isn't ideal. There are no additional images though.

Reply:
Also the order presented in conda-forge-pinning doesn't seem to matter when re-rendering the feedstock.

Reply:
I can fix this issue by making a tweak to conda-smithy. ( https://github.com/conda-forge/conda-smithy/pull/1176 ) It's a little hacky, but it does work. WDYT?

Reply:
Should add something is sorting `docker_image`'s values. I'm not sure if it is `conda-smithy` or `conda-build`.

Reply:
I think it's https://github.com/conda-forge/conda-smithy/blob/6b31b680b3d35b5b164a8ea619459782d004c126/conda_smithy/configure_feedstock.py#L114

Sorting was added there to have stable rerenders.

Reply:
Yeah that's what I figured. Was playing with that.

Maybe we can enforce an explicit sorting order here? Though am also ok with the conda-smithy PR above (if you are too).

Reply:
I'm okay with the conda-smithy PR, but does that work? From your previous comment, I thought it didn't work.

Reply:
It does work. We might want to come up with something more robust in the long run, but it's ok for now.